rasterio/_io.pyx

```diff
@@ -1745,18 +1746,19 @@ cdef class RasterUpdater(RasterReader):
 
         specifying a raster subset to write into.
         """
-        cdef void *hband
-        cdef void *hmask
-        if self._hds == NULL:
-            raise ValueError("can't write closed raster file")
-        hband = _gdal.GDALGetRasterBand(self._hds, 1)
-        if hband == NULL:
-            raise ValueError("NULL band mask")
-        if _gdal.GDALCreateMaskBand(hband, 0x02) != 0:
-            raise RuntimeError("Failed to create mask")
-        hmask = _gdal.GDALGetMaskBand(hband)
-        if hmask == NULL:
-            raise ValueError("NULL band mask")
+        cdef void *hband = NULL
+        cdef void *hmask = NULL
+
+        hband = self.band(1)
+        if GDALError.none != _gdal.GDALCreateMaskBand(hband, 0x02):
+            raise RasterioIOError("Failed to create mask.")
+
+        try:
+            with CPLErrors() as cple:
+                hmask = _gdal.GDALGetMaskBand(hband)
+                cple.check()
+        except:
+            raise
         log.debug("Created mask band")
```

Comment:
* In what cases do we raise raw `CPLE_*` exceptions versus wrapped in e.g. `RasterioIOError`?
* In some cases we have some cleanup/destroy calls and others we don't. What can we do to make sure we're not missing any potential memory leaks? To be cautious, maybe a comment is necessary to explain why we're *not* doing any cleanup.